tests/common/osutil/test_factory.py

```diff
@@ -0,0 +1,242 @@
+from azurelinuxagent.common.osutil.debian import DebianOSUtil, DebianOS8Util
+from azurelinuxagent.common.osutil.freebsd import FreeBSDOSUtil
+from azurelinuxagent.common.osutil.openbsd import OpenBSDOSUtil
+from azurelinuxagent.common.osutil.redhat import RedhatOSUtil, Redhat6xOSUtil
+from azurelinuxagent.common.osutil.suse import SUSEOSUtil, SUSE11OSUtil
+from azurelinuxagent.common.osutil.ubuntu import UbuntuOSUtil, Ubuntu12OSUtil, Ubuntu14OSUtil, \
+    UbuntuSnappyOSUtil, Ubuntu16OSUtil, Ubuntu18OSUtil
+from azurelinuxagent.common.osutil.alpine import AlpineOSUtil
+from azurelinuxagent.common.osutil.bigip import BigIpOSUtil
+from azurelinuxagent.common.osutil.gaia import GaiaOSUtil
+from azurelinuxagent.common.osutil.iosxe import IosxeOSUtil
+from azurelinuxagent.common.osutil.openwrt import OpenWRTOSUtil
+from tests.tools import *
+
+
+class DefaultOsUtilTestCase(AgentTestCase):
+
+    def setUp(self):
+        AgentTestCase.setUp(self)
+        self.mock__os_environ = patch.dict("os.environ", {'TRAVIS': 'false'})
+        self.mock__os_environ.start()
+
+    def tearDown(self):
+        self.mock__os_environ.stop()
+
+        AgentTestCase.tearDown(self)
+
+    def test_get_osutil_in_travis_environment_it_should_return_trusty(self):
+        mock_os_environ = {
+            'TRAVIS': 'true',
+            '_system_name': 'Ubuntu',
+            '_system_version': '14.04',
+            'TRAVIS_DIST': 'trusty',
+        }
+        with patch.dict("os.environ", mock_os_environ):
+            ret = get_osutil()
+            self.assertTrue(type(ret) == Ubuntu14OSUtil)
+
+    def test_get_osutil_in_travis_environment_it_should_return_xenial(self):
+        mock_os_environ = {
+            'TRAVIS': 'true',
+            '_system_name': 'Ubuntu',
+            '_system_version': '16.04',
+            'TRAVIS_DIST': 'xenial',
+        }
+        with patch.dict("os.environ", mock_os_environ):
+            ret = get_osutil()
+            self.assertTrue(type(ret) == Ubuntu16OSUtil)
+
+    def test_get_osutil_in_travis_environment_it_should_raise_and_catch(self):
+        # Leave out necessary environment variables to fail the retrieval and fall back to the given parameters
+        mock_os_environ = {
+            'TRAVIS': 'true'
+        }
+        with patch.dict("os.environ", mock_os_environ, clear=True):
+            ret = get_osutil(distro_name="debian",
+                             distro_version="8")
+            self.assertTrue(type(ret) == DebianOS8Util)
+
+    @patch("azurelinuxagent.common.logger.warn")
+    def test_get_osutil_it_should_return_default(self, patch_logger):
+        ret = get_osutil(distro_name="",
+                         distro_code_name="",
+                         distro_version="",
+                         distro_full_name="")
+        self.assertTrue(type(ret) == DefaultOSUtil)
+        self.assertEquals(patch_logger.call_count, 1)
+
+    def test_get_osutil_it_should_return_ubuntu(self):
```

Comment:
Can you move the Distro specific tests to its own classes? Keeping them in `DefaultOsUtilTestCase` is not the correct place.

Reply:
Misnomer, updated the name to reflect the scope of the tests.